### DISCUSSION TITLE
Avoiding interpolation for missing values for geom_line

### DISCUSSION THREAD

**Discussion by v--** [has2k1/plotnine#540] [Q&A] [answered]
I have a daily time series and I want geom_line to only connect two data points if they belong to the same streak of days, i.e. if there is no day between them with missing data. [This Stack Overflow question](https://stackoverflow.com/questions/61810857/how-to-avoid-the-connection-lines-in-geom-line-or-geom-path-when-there-is-no-dat) discusses something similar for ggplot2. Is there a way to achieve this using plotnine?

For example, is there a way to adapt the following code to produce a plot that does not attempt to connect 2 and 7?
```python
import plotnine as p9
import pandas as pd


df = pd.DataFrame(
    data=[(x, x ** 2) for x in range(10) if x < 3 or x > 6],
    columns=['x', 'y']
)

p9.ggplot(df) + p9.aes(x='x', y='y') + p9.geom_line()
```

**Comment by has2k1** [comment, marked as answer]
geom_line connects lines that belong to the same group. So you have to manipulate the data such that lines in the same streak are in the same group.

```python
import plotnine as p9
import pandas as pd

df = pd.DataFrame(
    data=[(x, x ** 2) for x in range(10) if x < 3 or x > 6],
    columns=['x', 'y']
)

# By walking along the difference between two consecutive values, we can
# categorise the streaks i.e. whenever the difference is not 1 then we have
# a new group.
group = []
g = 0
for d in df['x'].diff():
    if d != 1:
        g += 1   
    group.append(g)
    
df['group'] = group

p9.ggplot(df) + p9.aes(x='x', y='y', group='group') + p9.geom_line()
```

![streaks](https://user-images.githubusercontent.com/780341/146075185-25610173-a338-45d2-ad01-bdd026310a84.png)

**Reply by v--** [reply]
This is a much more elegant solution than I hoped for. Thank you. I very much appreciate your work on this project.

PS: Does the project accept donations?

**Reply by has2k1** [reply]
There isn't an easy way to take donations. Though [someone](https://github.com/has2k1/plotnine/issues/509#issuecomment-835786074) has donated by bank account through [wise.com](https://wise.com/).
